src/coreclr/tools/Common/Internal/Metadata/NativeFormat/NativeFormatReaderGen.cs

```diff
@@ -149,9 +149,9 @@ internal ByReferenceSignatureHandle(Handle handle) : this(handle._value)
 
         internal ByReferenceSignatureHandle(int value)
         {
-            HandleType hType = (HandleType)(value >> 24);
+            HandleType hType = (HandleType)((uint)value >> 25);
             Debug.Assert(hType == 0 || hType == HandleType.ByReferenceSignature || hType == HandleType.Null);
```

Comment:
```suggestion
            Debug.Assert(hType == 0 || hType == HandleType.ByReferenceSignature);
```
(somewhat unrelated) `HandleType.Null` is 0, so this is redundant.